### DISCUSSION TITLE
Add enums for router

### DISCUSSION THREAD

**Discussion by IgnisDa** [nuxt/nuxt#18126] [Ideas]
Using the router programmatically often results in errors since it uses strings.

```ts
router.push({name: 'accounts-login'})
```

Instead we can generate an `enum` which contains these routes as the variants. 

```ts
// somewhere in `.nuxt`
enum Routes {
  INDEX
  ACCOUNTS_LOGIN
  ACCOUNTS_REGISTER
  TEST
}
```
So that we can do the following:
```ts
// in your project
import { Routes } from '#app'

router.push({name: Routes.ACCOUNTS_LOGIN}) 
```

This will ensure that these errors are caught by the typescript compiler.

Here is a little sample implementation as a module:

```ts
import { defineNuxtModule, extendPages } from "@nuxt/kit";
import { toSnakeCase } from "js-convert-case";

export default defineNuxtModule({
  setup() {
    extendPages((pages) => {
      pages.map((e) => console.log(toSnakeCase(e.name).toUpperCase()));
    });
  },
});
```

**Comment by IgnisDa** [comment]
I can make a PR for the same if i am given a little guidance where to write this code.

**Comment by gnuser** [comment]
Test comment to verify Feishu webhook.

**Comment by angelhdzmultimedia** [comment]
If what you suggest is that the `NuxtPage` `name` field which is typed as `string | undefined` be changed to `string | undefined | Record<string, string | number>` (there's no type of generic enum),
I think the errors you got is because you defined your enum as a number enum, and not as a string enum!

```ts
enum Routes {
  INDEX
  ACCOUNTS_LOGIN
  ACCOUNTS_REGISTER
  TEST
}
```

the values for those keys are numbers, starting from 0!

```ts
router.push({name: Routes.ACCOUNTS_LOGIN})  // Error, ACCOUNTS_LOGIN is typed as `number` by default!!!
```

For it to be string you need to do this:

```ts
enum Routes {
  INDEX = 'INDEX',
  ACCOUNTS_LOGIN = 'ACCOUNTS_LOGIN ',
  ACCOUNTS_REGISTER = 'ACCOUNTS_REGISTER ',
  TEST = 'TEST '
}
```

```ts
router.push({name: Routes.ACCOUNTS_LOGIN})  // No errors!!! ACCOUNTS_LOGIN is now typed as `string`
```

Note: You forgot "," commas to delimit your enum fields.
